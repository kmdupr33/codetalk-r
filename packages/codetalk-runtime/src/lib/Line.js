import React from "react";
import PropTypes from "prop-types";
import SyntaxHighlighter from "react-syntax-highlighter";

const Line = ({ input }) => {
  console.log(`Line: ${JSON.stringify(input)}`);
  // Line: [["#","Hello world"]]
  // Line: [["##"," goodbye"]]
  // Line: ["idk why you say goodbye"]
  if (!input.some(element => Array.isArray(element))) {
    return <p>{input.join("")}</p>;
  }
  if (input[0][0] === "#") {
    return <h1>{input[0][1]}</h1>;
  }
  if (input[0][0] === "##") {
    return <h2>{input[0][1]}</h2>;
  }
  if (input[0][0] === "```") {
    const [_, language, code] = input[0];
    return <SyntaxHighlighter language={language}>{code}</SyntaxHighlighter>;
  }
  return input;
};

Line.propTypes = {
  input: PropTypes.array.isRequired
};

export default Line;
