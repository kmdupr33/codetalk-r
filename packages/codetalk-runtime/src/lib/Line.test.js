import React from "react";
import Line from "./Line";
import renderer from "react-test-renderer";

describe("Line", () => {
  it("handles a simple title", () => {
    const tree = renderer
      .create(<Line input={[["#", "Hello world"]]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("handles a simple subtitle", () => {
    const tree = renderer.create(<Line input={[["##", "goodbye"]]} />);
    expect(tree).toMatchSnapshot();
  });
  it("handles simple text", () => {
    const tree = renderer.create(<Line input={["idk why you say goodbye"]} />);
    expect(tree).toMatchSnapshot();
  })
  it("handles a newline", () => {
    
  });
});
