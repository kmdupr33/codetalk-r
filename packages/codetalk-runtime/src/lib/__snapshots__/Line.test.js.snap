// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line handles a simple subtitle 1`] = `
<h2>
  goodbye
</h2>
`;

exports[`Line handles a simple title 1`] = `
<h1>
  Hello world
</h1>
`;

exports[`Line handles simple text 1`] = `
<p>
  idk why you say goodbye
</p>
`;
